src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.cs

```diff
@@ -0,0 +1,401 @@
+                            else
+                            {
+                                if ((attribute.AttributeType == PlatformAttributeType.ObsoletedInOSPlatformAttribute ||
+                                    attribute.AttributeType == PlatformAttributeType.RemovedInOSPlatformAttribute) &&
+                                    AttributeVersionsMatch(attribute.AttributeType, attribute.Version, info.Version))
+                                {
+                                    return true;
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            return false;
+        }
+
+        private static void ReportDiagnosticsForAll(PooledDictionary<IOperation, ImmutableArray<PlatformAttributeInfo>> platformSpecificOperations, OperationBlockAnalysisContext context)
+        {
+            foreach (var platformSpecificOperation in platformSpecificOperations)
+            {
+                ReportDiagnostics(platformSpecificOperation.Key, platformSpecificOperation.Value.FirstOrDefault(), context);
+            }
+        }
+
+        private static void ReportDiagnostics(IOperation operation, PlatformAttributeInfo attribute, OperationBlockAnalysisContext context)
+        {
+            var operationName = GetOperationSymbol(operation)?.Name;
+
+            context.ReportDiagnostic(operation.CreateDiagnostic(SelectRule(attribute.AttributeType),
+                operationName ?? string.Empty, attribute.PlatformName, attribute.Version.ToString()));
+        }
+
+        private static ISymbol? GetOperationSymbol(IOperation operation)
+            => operation switch
+            {
+                IInvocationOperation iOperation => iOperation.TargetMethod,
+                IObjectCreationOperation cOperation => cOperation.Constructor,
+                IFieldReferenceOperation fOperation => IsWithinConditionalOperation(operation) ? null : fOperation.Field,
+                IMemberReferenceOperation mOperation => mOperation.Member,
+                _ => null,
+            };
+
+        private static void AnalyzeOperation(IOperation operation, OperationAnalysisContext context,
+            PooledDictionary<IOperation, ImmutableArray<PlatformAttributeInfo>> platformSpecificOperations)
+        {
+            using var builder = ArrayBuilder<PlatformAttributeInfo>.GetInstance();
+            var symbol = GetOperationSymbol(operation);
+
+            if (symbol != null)
+            {
+                var attributes = FindAllPlatformAttributesApplied(symbol.GetAttributes(), symbol.ContainingType);
+
+                foreach (AttributeData attribute in attributes)
+                {
+                    if (PlatformAttributeInfo.TryParsePlatformAttributeInfo(attribute, out PlatformAttributeInfo parsedAttribute))
+                    {
+                        if (!IsSuppressedByAttribute(parsedAttribute, context.ContainingSymbol))
```

Comment:
I think you need to restructure the attribute walking algorithm here. Your current algorithm walks and parses the whole containing symbol attribute chain of `context.ContainingSymbol` for every attribute applied to every platform specific symbol used/invoked in the operation block. Instead, you should compute the suppressing attributes for `context.ContainingSymbol` just once and create a cache/dictionary that can answer `IsSuppressedByAttribute` in O(1) without doing any computation.

Reply:
Yes that is on the list